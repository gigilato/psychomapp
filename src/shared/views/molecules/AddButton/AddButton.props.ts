import { IPressableIconProps } from '$molecules/Button'

export interface IAddButtonProps extends Omit<IPressableIconProps, 'icon'> {
  handleTabBar?: boolean
}
