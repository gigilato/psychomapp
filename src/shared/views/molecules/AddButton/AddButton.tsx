import { GestureResponderEvent } from 'react-native'

import { impactAsync } from '$libs/haptics'
import { IPressableIconProps, PressableIcon } from '$molecules/Button'

export const AddButton = ({ onPress: _onPress, ...props }: Omit<IPressableIconProps, 'icon'>) => {
  const onPress = (event: GestureResponderEvent) => {
    impactAsync()
    _onPress?.(event)
  }
  return (
    <PressableIcon
      animation="scale"
      shadow
      icon="plus"
      iconSize={24}
      iconClassName="text-white-classic"
      className="absolute right-l bottom-xl rounded-full bg-primary-classic p-m"
      onPress={onPress}
      {...props}
    />
  )
}
