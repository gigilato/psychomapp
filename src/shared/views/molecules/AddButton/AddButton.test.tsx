import { render } from '@testing-library/react-native'

import { AddButton } from './AddButton'

describe('AddButton', () => {
  it('should render correctly', async () => {
    const component = render(<AddButton ID="add" />)
    expect(component).toMatchSnapshot()
  })
})
