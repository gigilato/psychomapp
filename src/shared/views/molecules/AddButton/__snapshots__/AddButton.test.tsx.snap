// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  collapsable={false}
  focusable={true}
  hitSlop={5}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#FFFCF9",
      },
      {
        "shadowColor": "rgba(0, 0, 0, 0.1)",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 6,
      },
      {
        "shadowColor": "#D3D3D3",
      },
      {
        "position": "absolute",
      },
      {
        "right": 24,
      },
      {
        "bottom": 32,
      },
      {
        "borderBottomLeftRadius": 9999,
        "borderBottomRightRadius": 9999,
        "borderTopLeftRadius": 9999,
        "borderTopRightRadius": 9999,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID="add"
>
  <Text />
</View>
`;
